model/legacy-private/src/main/java/org/keycloak/storage/UserStorageManager.java

```diff
@@ -340,7 +340,7 @@ public UserModel getUserByEmail(RealmModel realm, String email) {
         if (user != null) {
             user = importValidation(realm, user);
             // Case when email was changed directly in the userStorage and doesn't correspond anymore to the email from local DB
-            if (email.equalsIgnoreCase(user.getEmail())) {
+            if (user != null && email.equalsIgnoreCase(user.getEmail())) {
```

Comment:
Maybe it would be correct to return `null` here, if `importValidation()` returns `null`?

Reply:
Sorry, I missed this question before. I don't think we want to return `null` in that case. If `importValidation` returns `null` it means the user from Keycloak database is invalid and was removed. Returning `null` in that situation would mean we say Keycloak does not know the user by this email, however, this is not true and it can be still present in an external provider. We need to query the user again.

Reply:
As I wrote, we DO NOT want to return `null` when validation fails, because some provider may still contain the user.

For example:
1. Some provider contains `user1` and it is synced to KC DB
2. The `user1` is changed in that provider (Keycloak DB is not updated). 
3. If we query this user by email, the validation will fail because the user was changed
4. Returning `null` in that case would be wrong because the provider still contains the user. We need to reload the user from the provider (or any other provider knowing user by that email).